service-base/src/main/java/org/eclipse/hono/service/command/CommandConnection.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.command;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.proton.ProtonDelivery;
+import org.eclipse.hono.client.HonoClient;
+
+import java.util.Map;
+
+/**
+ * A connection between an Adapter and the AMQP 1.0 network to receive commands and send a response.
```

Comment:
Maybe add the word `bidirectional` at the beginning? Would clarify it for me ...